.github/workflows/release-preview.yml

```diff
@@ -0,0 +1,197 @@
+        run: |
+          git push -f origin 5.x-preview
+
+  run_matomo_tests:
+    needs: [prepare_preview]
+    uses: ./.github/workflows/matomo-tests.yml
+    with:
+      is_preview: true
+
+  release_preview_version:
+    needs: [tests, prepare_preview]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          lfs: false
+          fetch-tags: true
+          ref: '5.x-preview'
+
+      - name: Import GPG key
+        if: steps.changes.outputs.do_release == '1' && steps.version.outputs.has_new_version == '1'
+        id: import_gpg
+        run: |
+          echo "${{ secrets.GPG_CERTIFICATE }}" > $HOME/private.asc
+          gpg --import --batch --yes $HOME/private.asc
+          echo "default-cache-ttl 7200
+          max-cache-ttl 31536000
+          allow-preset-passphrase" > $HOME/.gnupg/gpg-agent.conf
+          keygrip=$(gpg --import --import-options show-only --with-keygrip $HOME/private.asc | grep "Keygrip" | grep -oP "([A-F0-9]+)" | head -1)
+          hexPassphrase=$( echo -n '${{ secrets.GPG_CERTIFICATE_PASS }}' | od -A n -t x1 -w100 | sed 's/ *//g' )
+          gpg-connect-agent "RELOADAGENT" /bye
+          gpg-connect-agent "PRESET_PASSPHRASE ${keygrip} -1 ${hexPassphrase}" /bye
+          gpg-connect-agent "KEYINFO ${keygrip}" /bye
+
+      - name: Create tag, build and publish release
+        id: tag
+        run: |
+          echo "Version to build: '${NEW_VERSION}'"
+
+          TAG_EXISTS=$( git tag --list "$NEW_VERSION" )
+          if [[ -n "$TAG_EXISTS" ]]
+          then
+            echo "A tag for $tag_exists already exists."
+            exit 1
+          fi
+
+          echo "Creating a tag for $NEW_VERSION"
+
+          git tag $NEW_VERSION
+          git push origin tags/$NEW_VERSION
+
+          body="## Matomo ${version} (Pre-release)
+
+          We recommend to read [this FAQ](https://matomo.org/faq/how-to-update/faq_159/) before using a pre-release in a production environment.
+
+          Please use the attached archives for installing or updating Matomo.
+          The source code download is only meant for developers and will require extra work to install it.
+           - Latest stable production release can be found at https://matomo.org/download/ ([learn more](https://matomo.org/docs/installation/)) (recommended)
+           - Beta and Release Candidate releases can be found at https://builds.matomo.org/ ([learn more](https://matomo.org/faq/how-to-update/faq_159/))"
+
+          echo "version=$NEW_VERSION" >> $GITHUB_OUTPUT
+          echo 'body<<EOF' >> $GITHUB_OUTPUT
+          echo "$body" >> $GITHUB_OUTPUT
+          echo 'EOF' >> $GITHUB_OUTPUT
+
+          cd $GITHUB_WORKSPACE
+          chmod 755 ./.github/scripts/*.sh
+          ./.github/scripts/build-package.sh $NEW_VERSION
+        shell: bash
+        env:
+          NEW_VERSION: ${{ needs.prepare_preview.outputs.new_version }}
+
+      - uses: ncipollo/release-action@v1
```

Comment:
We probably should pin that to a specific commit instead of a movable tag. From what I have found this is at least done on in all other release workflows.

Reply:
Yeah, I noticed we have it pinned to a hash in other pipelines and thought that was done before there was a stable version. I can change this to be pinned to the same version as other actions, that's fine.